### ISSUE TITLE
Papercups down on Heroku - node 18

### ISSUE THREAD

**Issue by durdevic** [papercups-io/papercups#1024]
First of, thanks for keeping this repo available for all of us, I appreciate it a lot!

Quick question for some of the maintainers, would it be possible to update the code to be able to be deployed on heroku-20, since the heroku-18 is depricated. Also for a possible redeploy on app.papercups.io

https://help.heroku.com/X5OE6BCA/heroku-18-end-of-life-faq

I tried building my own version but I also got an error there and not sure how to move forward from there. 

Any help would be amazing 🎉

**Comment by ericdude4** [comment]
Any update here? Papercups down on my app :(

**Comment by durdevic** [comment]
I moved to crisp.chat 

If ppcups is back I'll think about switching back, but can't wait for that. 

They made a great work, great that it lasted for so long for free

**Comment by ronaldsantos97s** [comment]
I noticed Papercups is down on my app too. I'd love for a redeploy on app.papercups.io
![image (15)](https://github.com/papercups-io/papercups/assets/172663116/92d4413a-8f1e-4eb1-af9c-c004d0faeecb)

Do you know if we can self-host Papercups such that we don't need to ping Papercup's servers? If so, are there any requirements to self-host?

Thank you!

**Comment by bijitchakraborty** [comment]
> I noticed Papercups is down on my app too. I'd love for a redeploy on app.papercups.io ![image (15)](https://private-user-images.githubusercontent.com/172663116/339562252-92d4413a-8f1e-4eb1-af9c-c004d0faeecb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3NDAxODUsIm5iZiI6MTcxOTczOTg4NSwicGF0aCI6Ii8xNzI2NjMxMTYvMzM5NTYyMjUyLTkyZDQ0MTNhLThmMWUtNGViMS1hZjljLWMwMDRkMGZhZWVjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYzMFQwOTMxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZmI0YmU3OWZkMDllNzM5NTFlYzgzYTUzYjA3NTkzZDY4ZDE0OTY5OGYxZWU3NzBhNTg4NzRmOWMwZWQyZDI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ARFWXPfkaTdT96EfAbCSRb6IAF20ILpvoYqSJ_ijrUg)
> 
> Do you know if we can self-host Papercups such that we don't need to ping Papercup's servers? If so, are there any requirements to self-host?
> 
> Thank you!

Same issue. Isn't it possible to host it ourselves? What kind of FOSS project is this if we have to rely on them for their service? Have you found any solution?

**Comment by identang** [comment]
> I noticed Papercups is down on my app too. I'd love for a redeploy on app.papercups.io ![image (15)](https://private-user-images.githubusercontent.com/172663116/339562252-92d4413a-8f1e-4eb1-af9c-c004d0faeecb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNTQwNDcsIm5iZiI6MTcyMDA1Mzc0NywicGF0aCI6Ii8xNzI2NjMxMTYvMzM5NTYyMjUyLTkyZDQ0MTNhLThmMWUtNGViMS1hZjljLWMwMDRkMGZhZWVjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNFQwMDQyMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZTEzMDcyZGQ2NmIzMGJiYWQ0NjRmYWY4YmU1NWQ0ZDk2MmVkMWY5OGNiZDBlMWU1Nzk5MTQ2OWExMGYzYjE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.g-ljLnhmaOeF8QYAOU61H4Aqo3pqEQT2vKmNeA8NKFc)
> 
> Do you know if we can self-host Papercups such that we don't need to ping Papercup's servers? If so, are there any requirements to self-host?
> 
> Thank you!

Yeah, Papercups is also down on my app. I wish I can get it back as soon as possible. It would be great if we can self-host it.

**Comment by michalkow** [comment]
I knew Papercups was abandoned as a project, but I did not realize that React (Native) widgets depend on chat-widget.papercups.io domain...
Is there a way to just point to the self-hosted domain for the widgets as well? 
Is the chat-widget.papercups.io down temporarily or did it shut down for good this time?

**Comment by durdevic** [comment]
Guys already wrote in the README file, 
> Papercups is in maintenance mode. This means there won't be any major new features in the near future. We will still accept pull requests and conduct major bug fixes

So either someone picks this up or it's gone. 

Nice of them that they left it open, big appreciation for that tbh

**Comment by johncadengo** [comment]
@durdevic completely understood, but what I think people were hoping for was this:

> We will still [...] conduct major bug fixes

I think it's reasonable to see that if the project stops working, it's a bug.

**Comment by durdevic** [comment]
Fair, but the message is from Jan 2022, so 2 years ago. 

If your project failed and you needed to move on, I think it takes a huge chunk of your painful heart to still maintain something that doesn't bring you anything in that moment, at least would be for me.

Simply I understand the guys, thus, if something happens nice, regardless, I'm grateful for what they did so far.

**Comment by michalkow** [comment]
OK, I figured out how to self-host the chat widget iframe. If you get `Failed to load: chat-widget.papercups.io` here are the steps you need to take (requires a Vercel account):

1. Fork this repo: https://github.com/papercups-io/chat-window
2. Create new Vercel project from the git repo you just forked
3. Add `NODE_OPTIONS=--openssl-legacy-provider` to Environment Variables in project Settings
4. (Re)deploy from the git repo
5. Grab the domain URL assigned to you by Vercel and add it as `iframeUrlOverride` parameter to your `ChatWidget` component.

**Comment by 0x2b3bfa0** [comment]
Successfully deployed on Heroku following https://github.com/papercups-io/papercups/issues/1024#issuecomment-2210898044, but also had to patch [papercups-io/chat-window@0732357/package.json:10](https://github.com/papercups-io/chat-window/blob/07323571d902f0f5e4797a1d7d8365298d8c289c/package.json#L10) like this:

```diff
-     "start": "next start -p 8080"
+     "start": "next start -p $PORT"
```

**Comment by identang** [comment]
> OK, I figured out how to self-host the chat widget iframe. If you get `Failed to load: chat-widget.papercups.io` here are the steps you need to take (requires a Vercel account):
> 
> 1. Fork this repo: https://github.com/papercups-io/chat-window
> 2. Create new Vercel project from the git repo you just forked
> 3. Add `NODE_OPTIONS=--openssl-legacy-provider` to Environment Variables in project Settings
> 4. (Re)deploy from the git repo
> 5. Grab the domain URL assigned to you by Vercel and add it as `iframeUrlOverride` parameter to your `ChatWidget` component.

We didn't get 404 error anymore after deploying the code to Vercel and adding the link to `iframeUrlOverride` param. However, we ran into an error`Error: An account token is required to render the Papercups chat widget!` even we double checked we put the token as a param. Does anyone have the same issue here?

**Comment by 0x2b3bfa0** [comment]
I didn't have this issue. 🤔

**Comment by michalkow** [comment]
@identang Are you self-hosting Papercups App as well? Make sure that `accountId` and `baseUrl` props are set correctly.

**Comment by identang** [comment]
> @identang Are you self-hosting Papercups App as well? Make sure that `accountId` and `baseUrl` props are set correctly.

Yes, we are trying to self-hosting papercups as well. ~~I am sure about the correctness of both. We use the same `accountId` and `baseUrl` before it was down and it worked.~~ Let me double check it to make sure that.

**Comment by GildedHonour** [comment]
Calmaos :) Atención! 

I'm working on it; have rewritten some parts, refactored others. Some things don't work such as the bloody React part which I'm rewritting altogether. Once it's more or less done, I'll notify you.
